### PR TITLE
Readme rev

### PR DESCRIPTION
**Pull request by jbondpdx**
Major revision of README:

Reorganized, rewritten, type and provider information added, new parameters and private classes updated.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,64 +32,113 @@ Compatibility mode may fail. It may eat your MySQL server. PLEASE test it before
...
+you can make an entry like `thing => true`, `thing => value`, or `thing => "` in the hash. Alternatively, you can pass an array, as `thing => ['value', 'value2']`, or list each `thing => value` separately on separate lines. 
 
-You can just make an entry like `thing => true`, `thing => value`, or `thing => "` in the hash. You can also pass an array `thing => ['value', 'value2']` or even list each `thing => value` separately on separate lines. MySQL doesn't care if 'thing' is alone or set to a value; it'll happily accept both.  To keep an option out of the my.cnf file, e.g. when using override_options to revert to a default value, you can pass thing => undef.
-If an option needs multiple instances, you can pass an array. For example
+MySQL doesn't care if 'thing' is alone or set to a value; it happily accepts both. To keep an option out of the my.cnf file --- e.g., when using `override_options` to revert to a default value --- you can pass `thing => undef`.
 
-```puppet
+If an option needs multiple instances, you can pass an array. For example,
+
+`~~~
```
Comment (igalic) [inline README.md]: this ` breaks formatting

File: README.md [hunk 2]
```diff
@@ -44,64 +32,113 @@ Compatibility mode may fail. It may eat your MySQL server. PLEASE test it before
...
+All interaction for the server is done via `mysql::server`. To install the client, use `mysql::client`. To install bindings, use `mysql::bindings`.
 
-###Overrides
+###Customizing Server Options
 
-The hash structure for overrides in `mysql::server` is as follows:
+The hash structure for overrides in `mysql::server` can be structured like a hash in the my.cnf file, so:
 
-```puppet
+~~~
```
Comment (igalic) [inline README.md]: is there a reason you're changing this from the specific "```puppet" to the general "~~~"?

File: README.md (hunk 1 above)
Comment (jbondpdx) [inline README.md]: good catch, ty

File: README.md (hunk 2 above)
Comment (jbondpdx) [inline README.md]: Docs team decision for style and display consistency.

**Comment by jbondpdx** [comment]
PR updated with fixed typo.

**Comment by mhaskel** [comment]
\o/ Thanks @jbondpdx !

**Pull request merged**